lib/modules/datasource/maven/index.ts

```diff
@@ -249,61 +249,81 @@ export class MavenDatasource extends Datasource {
     }
 
     const cacheNs = 'datasource-maven:head-requests';
+    const cacheTimeoutNs = 'datasource-maven:head-requests-timeout';
     const cacheKey = `${repoUrl}${dependency.dependencyUrl}`;
-    const oldReleaseMap: ReleaseMap | undefined =
-      await packageCache.get<ReleaseMap>(cacheNs, cacheKey);
-    const newReleaseMap: ReleaseMap = oldReleaseMap ?? {};
-
-    if (!oldReleaseMap) {
-      const unknownVersions = Object.entries(releaseMap)
-        .filter(([version, release]) => {
-          const isDiscoveredOutside = !!release;
-          const isDiscoveredInsideAndCached = !is.undefined(
-            newReleaseMap[version]
-          );
-          const isDiscovered =
-            isDiscoveredOutside || isDiscoveredInsideAndCached;
-          return !isDiscovered;
-        })
-        .map(([k]) => k);
-
-      if (unknownVersions.length) {
-        let retryEarlier = false;
-        const queue = unknownVersions.map(
-          (version) => async (): Promise<void> => {
-            const pomUrl = await this.createUrlForDependencyPom(
-              version,
-              dependency,
-              repoUrl
-            );
-            const artifactUrl = getMavenUrl(dependency, repoUrl, pomUrl);
-            const release: Release = { version };
-
-            const res = await checkHttpResource(this.http, artifactUrl);
-
-            if (res === 'error') {
-              retryEarlier = true;
-            }
 
-            if (is.date(res)) {
-              release.releaseTimestamp = res.toISOString();
-            }
+    // Store cache validity as the separate flag.
+    // This allows both cache updating and resetting.
```

Comment:
I don't understand this - can you describe more?

Reply:
Just added more description to the comment